### PR TITLE
Make Initial Setup and its startup more robust

### PR DESCRIPTION
**Pull request by M4rtinK**
The main aim of this pull request is to make initial setup more robust by eliminating the possible race condition between the two Initial Setup units (_initial-setup-text.service_ and _initial-setup-graphical.service_). 

**How it works**
This is done by switching to a single unit named _initial-setup.service_, which either runs the GUI if it is installed or just the TUI if not. Having just one unit eliminates the race condition and generally simplifies things.

**Backward compatibility**
Since _initial-setup-text.service_ and _initial-setup-graphical.service_ units have been used on RHEL since 7.0 both unit files are still present for backward compatibility reasons. But internally both units actually just log a depreciation warning and start the _initial-setup.service_. The old behavior of only running the TUI if only the text unit is enabled is also emulated.

**Additional improvements**
The pull request also contains some related robustness & maintainability improvements:
* files not intended to be run manually are no longer in $PATH
* the s390 startup scripts are now in a separate directory
* improved logging for kickstart syntax errors
* Initial Setup prints version when starting up
* the Makefile now makes the release & build process easier and more robust

### CODE REVIEW

File: initial_setup/__init__.py [hunk 1]
```diff
@@ -0,0 +1,289 @@
...
+            # gui already imported (see above)
+
+            # Add addons to search paths
+            initial_setup.gui.InitialSetupGraphicalUserInterface.update_paths(self._addon_module_paths)
+
+            # Initialize the UI
+            log.debug("initializing GUI")
+            ui = initial_setup.gui.InitialSetupGraphicalUserInterface(None, None, PostInstallClass())
+        else:
+            # Import IS gui specifics
```
Comment (vpodzime) [inline initial_setup/__init__.py]: tui specifics

File: systemd/initial-setup-graphical.service [hunk 2]
```diff
@@ -1,17 +1,28 @@
...
 ConditionKernelCommandLine=!rd.live.image
+Requires=initial-setup.service
 
 [Service]
 Type=oneshot
+TimeoutSec=0
+StandardInput=tty
+StandardOutput=tty
+RemainAfterExit=yes
+ExecStartPre=/bin/kill -55 1
```
Comment (vpodzime) [inline systemd/initial-setup-graphical.service]: I think the signaling should stay just in the main ``.service`` file.

**Comment by vpodzime** [comment]
Looks good to me otherwise.

File: systemd/initial-setup-graphical.service (hunk 2 above)
Comment (M4rtinK) [inline systemd/initial-setup-graphical.service]: Good point.

**Comment by jkonecny12** [comment]
There is a small typo in commit "Make Initial Setup startup more robust (#1249598)":

The single unit file will run a shell script that checks if the GUI executable exists and runs it if it does. If **the(re)** is no GUI executable the TUI executable will be run.

File: initial_setup/__init__.py [hunk 3]
```diff
@@ -0,0 +1,289 @@
...
+        if self.gui_mode:
+            return "gui"
+        else:
+            return "tui"
+
+    def _parse_arguments(self):
+        """Parse command line arguments"""
+        # create an argparse instance
+        parser = argparse.ArgumentParser(prog="Initial Setup",
+                                         description="Initial Setup is can run during the first start of a newly installed"
```
Comment (jkonecny12) [inline initial_setup/__init__.py]: I guess this should be without *is*.

File: initial_setup/__init__.py [hunk 4]
```diff
@@ -0,0 +1,289 @@
...
+        # would start again next time the Initial Setup service is enabled
+        if external_reconfig and os.path.exists(RECONFIG_FILE):
+            log.debug("removing the reconfig file")
+            os.remove(RECONFIG_FILE)
+            log.debug("the reconfig file has been removed")
+
+    def run(self):
+        """Run Initial setup
+
+        :param bool gui_mode: if GUI should be used (TUI is the default)
```
Comment (jkonecny12) [inline initial_setup/__init__.py]: I don't know if using param in doctype is good idea when this is parameter passed in by self.gui_mode.

File: scripts/run-initial-setup [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+TEXT_UNIT_ENABLED=0
+GRAPHICAL_UNIT_ENABLED=0
+GUI_INSTALLED=0
+
+WINDOWMANAGER_SCRIPT="/usr/libexec/initial-setup/firstboot-windowmanager"
+
+START_GUI_COMMAND="/bin/xinit ${WINDOWMANAGER_SCRIPT} ${IS_GRAPHICAL} --no-stdout-log -- /bin/Xorg :9 -ac -nolisten tcp"
+
+# check if the text and graphical services are enabled
+# Note: These services are deprecated and only suppoirted for backward compatibility,
```
Comment (jkonecny12) [inline scripts/run-initial-setup]: suppoirted = *supported*

File: scripts/s390/initial-setup.csh [hunk 6]
```diff
@@ -0,0 +1,21 @@
+# initial-setup.csh
+
+set IS_EXEC = /lib/exec/initial-setup-text
+set IS_UNIT = initial-setup.service
+
+# the initial-setup-text.service is depreciated, use initial-setup.service instead
```
Comment (jkonecny12) [inline scripts/s390/initial-setup.csh]: depreciated == *deprecated*

**Comment by jkonecny12** [comment]
Other than the few above this looks good to me too.

**Comment by M4rtinK** [comment]
@jkonecny12 & @vpodzime Thanks for the review! :)

**Comment by M4rtinK** [comment]
Pushed manually.

**Pull request closed without merging**